src/js/modules/facetedsearch/update.ts

```diff
@@ -80,6 +80,18 @@ export default () => {
     );
   });
 
+  /**
+   * Pager links also scroll up
+   */
+  $('body').on('click', Theme.selectors.listing.pagerLink, (event) => {
+    event.preventDefault();
+    document.getElementById('js-product-list-top')?.scrollIntoView({block: 'start', behavior: 'auto'});
```

Comment:
Maybe use here `Theme.selectors.listing.listTop`

Reply:
```
src/js/constants/selectors-map.ts:  listTop: '#js-product-list-top',
```
Those are in jQuery format.. should there be a `listTopId: 'js-product-list-top'`

Reply:
Or 

```diff
--- a/src/js/modules/facetedsearch/update.ts
+++ b/src/js/modules/facetedsearch/update.ts
@@ -85,7 +85,8 @@ export default () => {
    */
   $('body').on('click', Theme.selectors.listing.pagerLink, (event) => {
     event.preventDefault();
-    document.getElementById('js-product-list-top')?.scrollIntoView({block: 'start', behavior: 'auto'});
+    /* selectors are in jQuery format, slice the leading # character out */
+    document.getElementById(Theme.selectors.listing.listTop.slice(1))?.scrollIntoView({block: 'start', behavior: 'auto'});
     prestashop.emit(
       events.updateFacets,
       $(event.target)?.closest('a')?.get(0)?.getAttribute('href'),
```

but this feels a bit dirty..

Reply:
> ```
> src/js/constants/selectors-map.ts:  listTop: '#js-product-list-top',
> ```
> 
> Those are in jQuery format.. should there be a `listTopId: 'js-product-list-top'`

Well, first of all I'm not sure it should even be an ID. Any `js-...` selector should be a class.

And what do you mean by _jQuery format_? They are simple selectors :)
Just use `document.querySelector(Theme.selectors.listing.listTop)` instead of `document.getElementById('js-product-list-top')`

Reply:
well, css selector (where `#` denotes id) which jQuery uses everywhere..

vanilla js getElementById takes only the id not the css selector. 

getElementById is faster than querySelector (not by much but it is by definition).

I'll change it to use querySelector

Reply:
Some say [querySelector is faster](https://www.reddit.com/r/learnjavascript/comments/i0f5o8/performance_of_getelementbyid_vs_queryselector/?rdt=42342) 😉 Doesn't matter nowadays really, but just for consistency we should use `querySelector` and `querySelectorAll`. Especially since we're using selectors-map.

Reply:
yup, it was changed here [4a29900](https://github.com/PrestaShop/hummingbird/pull/565/commits/4a299009dbe9758aa6a5da22a65f2921bc252f14)